### PR TITLE
Do not apply redirection for delete table scans

### PR DESCRIPTION
**Pull request by sopel39**

### CODE REVIEW

**Comment by raunaqmorarka** [comment]
We can have a test `TestApplyTableScanRedirection#doesNotFireIfDeleteTableScan`

**Comment by sopel39** [comment]
added test

**Review by raunaqmorarka — APPROVED** [review]

**Review by martint — APPROVED** [review]

File: core/trino-main/src/test/java/io/trino/sql/planner/iterative/rule/TestApplyTableScanRedirection.java [hunk 1]
```diff
@@ -105,6 +105,30 @@ public void testDoesNotFire()
         }
     }
 
+    @Test
+    public void testDoesNotFireForDeleteTableScan()
```
Comment (findepi) [inline core/trino-main/src/test/java/io/trino/sql/planner/iterative/rule/TestApplyTableScanRedirection.java:109]: Would it be possible to have an integration test as well?
there can be other problem(s) in delete flow; it's kind of special

Comment (sopel39) [reply core/trino-main/src/test/java/io/trino/sql/planner/iterative/rule/TestApplyTableScanRedirection.java:109]: > Would it be possible to have an integration test as well?

We don't have a connector implementation yet that supports redirections and deletes

**Pull request merged**